sdk/synapse/Azure.Analytics.Synapse.Spark/samples/Sample2_ExecuteSparkStatement.md

```diff
@@ -0,0 +1,200 @@
+
+It is hoped to be temporary.
+
+```C# Snippet:TemporarySparkSupportCode
+ private const string Error = "error";
+ private const string Dead = "dead";
+ private const string Success = "success";
+ private const string Killed = "killed";
+ private const string Idle = "idle";
+
+public static List<string> SessionSubmissionFinalStates = new List<string>
+ {
+     Idle,
+     Error,
+     Dead,
+     Success,
+     Killed
+ };
+
+ public static SparkSession PollSparkSession(
+     SparkSessionClient client,
+     SparkSession session,
+     IList<string> livyReadyStates = null)
+ {
+     if (livyReadyStates == null)
+     {
+         livyReadyStates = SessionSubmissionFinalStates;
+     }
+
+     return Poll(
+         session,
+         s => s.Result.ToString(),
+         s => s.State,
+         s => client.GetSparkSession(s.Id, true),
+         livyReadyStates);
+ }
+
+ private const string Starting = "starting";
+ private const string Waiting = "waiting";
+ private const string Running = "running";
+ private const string Cancelling = "cancelling";
+
+ private static List<string> ExecutingStates = new List<string>
+ {
+     Starting,
+     Waiting,
+     Running,
+     Cancelling
+ };
+
+ private static SparkStatement PollSparkStatement(
+     SparkSessionClient client,
+     int sessionId,
+     SparkStatement statement)
+ {
+     return Poll(
+         statement,
+         s => null,
+         s => s.State,
+         s => client.GetSparkStatement(sessionId, s.Id),
+         ExecutingStates,
+         isFinalState: false);
+ }
+
+ private static T Poll<T>(
+     T job,
+     Func<T, string> getJobState,
+     Func<T, string> getLivyState,
+     Func<T, T> refresh,
+     IList<string> livyReadyStates,
+     bool isFinalState = true,
+     int pollingInMilliseconds = 0,
+     int timeoutInMilliseconds = 0,
+     Action<T> writeLog = null)
+ {
+     var timeWaitedInMilliSeconds = 0;
+     if (pollingInMilliseconds == 0)
+     {
+         pollingInMilliseconds = 5000;
+     }
+
+     while (IsJobRunning(getJobState(job), getLivyState(job), livyReadyStates, isFinalState))
+     {
+         if (timeoutInMilliseconds > 0 && timeWaitedInMilliSeconds >= timeoutInMilliseconds)
+         {
+             throw new TimeoutException();
+         }
+
+         writeLog?.Invoke(job);
+         //TestMockSupport.Delay(pollingInMilliseconds);
+         System.Threading.Thread.Sleep(pollingInMilliseconds);
+         timeWaitedInMilliSeconds += pollingInMilliseconds;
+
+         // TODO: handle retryable excetpion
+         job = refresh(job);
+     }
+
+     return job;
+ }
+
+ private static bool IsJobRunning(string jobState, string livyState, IList<string> livyStates, bool isFinalState = true)
+ {
+     if ("Succeeded".Equals(jobState, StringComparison.OrdinalIgnoreCase)
+         || "Failed".Equals(jobState, StringComparison.OrdinalIgnoreCase)
+         || "Cancelled".Equals(jobState, StringComparison.OrdinalIgnoreCase))
```

Comment:
It's a bit confusing that the constant definitions and the list sets are using lower-case, but the `if` block has hard-coded uppercase.    Since the list `Contains` is case-sensitive, it may be helpful to keep any hard-coded string checks consistent with that same casing.